### PR TITLE
libnss_nis: init at 3.2

### PR DESCRIPTION
**Pull request by BarrOff**
This PR adds the *libnss_nis* NSS module package.
It allows [NSS](https://en.wikipedia.org/wiki/Name_Service_Switch) to query [NIS](https://en.wikipedia.org/wiki/Network_Information_Service) for information.
More information is available on the [project's homepage](https://github.com/thkukuk/libnss_nis)

## Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
- [x] tested the package against a real world NIS system

### CODE REVIEW

File: pkgs/by-name/li/libnss_nis/package.nix [hunk 1]
```diff
@@ -0,0 +1,44 @@
...
+    rev = "v${version}";
+    hash = "sha256-dt5wL+v98Heg6395BOwNssXLXmoOKFnRXGqlOknYYPs=";
+  };
+
+  outputs = [ "out" ];
+
+  meta = {
+    description = "NSS module for glibc, to provide NIS support for glibc.";
+    changelog = "https://github.com/thkukuk/${pname}/blob/master/NEWS";
+    homepage = "https://github.com/thkukuk/${pname}";
```
Comment (FliegendeWurst) [inline pkgs/by-name/li/libnss_nis/package.nix:39]:
```suggestion
    changelog = "https://github.com/thkukuk/libnss_nis/blob/master/NEWS";
    homepage = "https://github.com/thkukuk/libnss_nis";
```

File: pkgs/by-name/li/libnss_nis/package.nix [hunk 2]
```diff
@@ -0,0 +1,44 @@
...
+    license = lib.licenses.lgpl21;
```
Comment (FliegendeWurst) [inline pkgs/by-name/li/libnss_nis/package.nix:40]:
```suggestion
    license = lib.licenses.lgpl21Plus;
```

**Review by FliegendeWurst — CHANGES_REQUESTED** [review]

File: pkgs/by-name/li/libnss_nis/package.nix (hunk 2 above)
Comment (BarrOff) [reply pkgs/by-name/li/libnss_nis/package.nix:40]: I am not really an expert with these legal things, but the [license file](https://github.com/thkukuk/libnss_nis/blob/master/COPYING) of the repo states LGPL 2.1.
Why do you suggest changing that to LGPL 2.1+?

Comment (FliegendeWurst) [reply pkgs/by-name/li/libnss_nis/package.nix:40]: The license header in each source file states that later versions are also applicable

Comment (BarrOff) [reply pkgs/by-name/li/libnss_nis/package.nix:40]: Oh, I didn't look that close at the license headers.

I have adjusted the license attribute.

File: pkgs/by-name/li/libnss_nis/package.nix [hunk 3]
```diff
@@ -0,0 +1,44 @@
...
+
+  buildInputs = [
+    libxcrypt
+    libnsl
+    libtirpc
+  ];
+
+  src = fetchFromGitHub {
+    owner = "thkukuk";
+    repo = pname;
```
Comment (FliegendeWurst) [inline pkgs/by-name/li/libnss_nis/package.nix:29]:
```suggestion
    repo = "libnss_nis";
```

File: pkgs/by-name/li/libnss_nis/package.nix [hunk 4]
```diff
@@ -0,0 +1,44 @@
...
+    rev = "v${version}";
+    hash = "sha256-dt5wL+v98Heg6395BOwNssXLXmoOKFnRXGqlOknYYPs=";
+  };
+
+  outputs = [ "out" ];
+
+  meta = {
+    description = "NSS module for glibc, to provide NIS support for glibc.";
```
Comment (FliegendeWurst) [inline pkgs/by-name/li/libnss_nis/package.nix:37]:
```suggestion
    description = "NSS module for glibc, to provide NIS support for glibc";
```

**Review by FliegendeWurst — CHANGES_REQUESTED** [review]

**Review by FliegendeWurst — APPROVED** [review]

File: pkgs/by-name/li/libnss_nis/package.nix (hunk 3 above)
Comment (BarrOff) [reply pkgs/by-name/li/libnss_nis/package.nix]: Done.

File: pkgs/by-name/li/libnss_nis/package.nix (hunk 4 above)
Comment (BarrOff) [reply pkgs/by-name/li/libnss_nis/package.nix]: Done

**Pull request merged**